src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/Models/ReindexJobRecord.cs

```diff
@@ -38,10 +39,53 @@ public class ReindexJobRecord : JobRecord
             LastModified = Clock.UtcNow;
 
             ResourceTypeSearchParameterHashMap = searchParametersHash;
-            MaximumConcurrency = maxiumumConcurrency;
-            MaximumNumberOfResourcesPerQuery = maxResourcesPerQuery;
-            QueryDelayIntervalInMilliseconds = queryDelayIntervalInMilliseconds;
-            TargetDataStoreUsagePercentage = targetDataStoreUsagePercentage;
+
+            // check for MaximumConcurrency boundary
+            if (maxiumumConcurrency > 10)
+            {
+                throw new BadRequestException(string.Format(Fhir.Core.Resources.InvalidReIndexParameterValue, nameof(MaximumConcurrency)));
+            }
+            else
+            {
+                MaximumConcurrency = maxiumumConcurrency;
+            }
+
+            // check for MaximumNumberOfResourcesPerQuery boundary
+            if (maxResourcesPerQuery < 1 || maxResourcesPerQuery > 5000)
+            {
+                throw new BadRequestException(string.Format(Fhir.Core.Resources.InvalidReIndexParameterValue, nameof(MaximumNumberOfResourcesPerQuery)));
+            }
+            else
+            {
+                MaximumNumberOfResourcesPerQuery = maxResourcesPerQuery;
+            }
+
+            // check for QueryDelayIntervalInMilliseconds boundary
+            if (queryDelayIntervalInMilliseconds < 5 || queryDelayIntervalInMilliseconds > 500000)
+            {
+                throw new BadRequestException(string.Format(Fhir.Core.Resources.InvalidReIndexParameterValue, nameof(QueryDelayIntervalInMilliseconds)));
+            }
+            else
+            {
+                QueryDelayIntervalInMilliseconds = queryDelayIntervalInMilliseconds;
+            }
+
+            // check for TargetDataStoreUsagePercentage boundary
+            if (targetDataStoreUsagePercentage < 0 || targetDataStoreUsagePercentage > 100)
```

Comment:
>< 0 [](http://example.com/codeflow?start=47&length=3)

What `targetDataStoreUsagePercentage` = 0 means if we allow 0?

Reply:
yes we do allow 0 for targetDataStoreUsagePercentage